api/client/proto/authservice.proto

```diff
@@ -977,6 +1019,89 @@ message DeleteWindowsDesktopRequest {
     string Name = 1 [ (gogoproto.jsontag) = "name" ];
 }
 
+// ChangePasswordWithTokenRequest defines a request to change one or more of a user's auth
+// credential.
+message ChangePasswordWithTokenRequest {
+    string TokenID = 1;
+    // SecondFactorToken is an otp value.
+    string SecondFactorToken = 2;
+    // Password is user's password.
+    bytes Password = 3;
+    // U2FRegisterResponse is U2F registration challenge response.
+    U2FRegisterResponse U2FRegisterResponse = 4;
+    // DeviceName is the name of the second factor device.
+    string DeviceName = 5;
+}
```

Comment:
I think this should be something in the lines of the message below:

```suggestion
message ChangePasswordWithTokenRequest {
    UserCredentials Credentials = 1;
    MFAAuthenticateResponse ExistingMFAResponse = 2;
    string NewDeviceName = 3;
    MFARegisterResponse NewMFARegisterResponse = 4;
}
```

I also think this message (and RPC) are misnamed - they change both password AND add a new device, which the name doesn't imply.

Reply:
renamed rpc and mesage to `ChangeAuthentication` (did not include `MFAAuthenticateResponse` and kept TokenID for now)